ipatests/test_integration/test_replication_layouts.py

```diff
@@ -178,3 +181,47 @@ def test_2_connected_topology_with_ca_kra(self):
         tasks.install_topo('double-circle', self.master, self.replicas, [],
                            setup_replica_cas=True, setup_replica_kras=True)
         self.replication_is_working()
+
+class TestReplicaInstallWithExistingEntry(LayoutsBaseTest):
+    """replica install might fail because of existing entry for replica like
+    `cn=ipa-http-delegation,cn=s4u2proxy,cn=etc,$SUFFIX` etc. The situation
+    may arise due to incorrect uninstall of replica.
+
+    https://pagure.io/freeipa/issue/7174"""
+
+    num_replicas = 1
+
+    def test_replica_install_with_existing_entry(self):
+        result = tasks.install_master(self.master)
+        assert result.returncode == 0
+        master = self.master
+        replica = self.replicas[0]
+        tf = NamedTemporaryFile()
+        ldif_file = tf.name
+        base_dn = "dc=%s" % (",dc=".join(replica.domain.name.split(".")))
+        #adding entry for replica on master so that master will have it before
+        # replica installtion begins and creates a situation for pagure-7174
+        entry_ldif = textwrap.dedent("""
+        dn: cn=ipa-http-delegation,cn=s4u2proxy,cn=etc,{base_dn}\n
+        changetype: modify\n
+        add: memberPrincipal\n
+        memberPrincipal: HTTP/{hostname}@{realm}\n
+
+        dn: cn=ipa-ldap-delegation-targets,cn=s4u2proxy,cn=etc,{base_dn}\n
+        changetype: modify\n
+        add: memberPrincipal\n
+        memberPrincipal: ldap/{hostname}@{realm}""").format(
+        base_dn=base_dn, hostname=replica.hostname,
+        realm=replica.domain.name.upper())
+        master.put_file_contents(ldif_file, entry_ldif)
+        arg = ['ldapmodify', '-h', master.hostname, '-p', '389',
+               '-D', str(master.config.dirman_dn), '-w',
+               master.config.dirman_password, '-f', ldif_file]
+        cmd = master.run_command(arg, raiseonerr=False)
+        assert cmd.returncode == 0, cmd.stdout_text
```

Comment:
Why are you using ``raiseonerr=False`` and then checking the result for error? How about you use the default and let the command fail with a nice exception.